modules/game/src/main/Metadata.scala

```diff
@@ -33,9 +33,12 @@ private[game] object Metadata:
 
 case class GameDrawOffers(white: Set[Ply], black: Set[Ply]):
 
+  var lastDrawColor: Option[Color] = None
```

Comment:
We should not use mutable variables in most (or all) cases, especially at public one like this one.

And the way it is being used is also wrong. `add` function in line 40 will return another `GameDrawOffers`, so line 41 just does nothing.

Reply:
Hi lenguyenthanh and thanks for the review :) !

I didn't know that copy was used that much in relation to case class in Scala, and that it is used that way instead of using mutable variables, good to know !

I changed my commit which I think should be much better now